components/email/src/Email.svelte

```diff
@@ -665,40 +675,40 @@
       message_id: _this.message_id,
     }).then(async (json) => {
       _this.message = json;
-      messageLoadStatus[0] = "loaded";
       if (FilesStore.hasInlineFiles(_this.message)) {
         const message = await getMessageWithInlineFiles(_this.message);
         _this.message = message;
       }
     });
   }
 
-  function handleDraftClick(event: MouseEvent, draftIndex: number) {
+  function handleDraftClick(event: MouseEvent, draft: Message) {
     event.stopImmediatePropagation();
-    dispatchDraftEvent(event, draftIndex);
+    dispatchDraftEvent(event, draft);
   }
 
-  function handleDraftKeypress(event: KeyboardEvent, draftIndex: number) {
+  function handleDraftKeypress(event: KeyboardEvent, draft: Message) {
     event.stopImmediatePropagation();
     if (event.code === "Enter") {
-      dispatchDraftEvent(event, draftIndex);
+      dispatchDraftEvent(event, draft);
     }
   }
 
-  function dispatchDraftEvent(event: UIEvent, draftIndex: number) {
-    if (activeThread.drafts[draftIndex]) {
+  function dispatchDraftEvent(event: UIEvent, draft: Message) {
+    if (draft) {
+      activeThread?.drafts?.forEach(
+        (threadDraft) => (threadDraft.active = threadDraft.id === draft.id),
+      );
+      draft.active = true;
       dispatchEvent("draftClicked", {
         event,
-        message: activeThread.drafts[draftIndex],
+        message: draft,
         thread: activeThread,
       });
       // Don't fetch message when thread is being passed manually
-      if (!_this.thread) {
-        fetchIndividualMessage(
-          draftIndex,
-          activeThread.drafts[draftIndex].id,
-        ).then((res) => {
-          activeThread.drafts[draftIndex].body = res;
+      if (!_this.thread && draft.id) {
+        fetchIndividualMessage(draft.id).then((res) => {
+          draft.body = res;
```

Comment:
Don't we need to update `activeThread.drafts` here?

Reply:
Currently all the usages of `dispatchDraftEvent` are having the `draft` parameter as reference.